test/specs/test-util.el

```diff
@@ -46,6 +46,35 @@
         (goto-char (point-min))
         (narrow-to-region (point-min) (point-min))
         (expect (buffer-string) :to-equal "")
-        (expect (flycheck-buffer-empty-p) :not :to-be-truthy)))))
+        (expect (flycheck-buffer-empty-p) :not :to-be-truthy))))
+
+  (describe "flycheck-buffer-saved-p"
+
+    (it "considers an unmodified buffer without backing file unsaved"
+      (with-temp-buffer
+        (expect (flycheck-buffer-saved-p) :not :to-be-truthy)))
+
+    (it "considers a modified buffer without backing file unsaved"
+      (with-temp-buffer
+        (set-buffer-modified-p t)
+        (expect (flycheck-buffer-saved-p) :not :to-be-truthy)))
+
+    (it "considers an unmodified buffer with backing file saved"
+      (spy-on 'file-exists-p :and-call-fake (lambda (&optional buffer) t))
+      (spy-on 'buffer-file-name :and-call-fake (lambda (&optional buffer) "test-buffer-name"))
```

Comment:
@rski Is there a reason why you're not using `:and-return-value "test-buffer-name"` here?